watertap/property_models/activated_sludge/tests/test_asm1_integration.py

```diff
@@ -0,0 +1,386 @@
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Tests for ASM1 proeprty and reaction packages.
+
+Verified againts results from:
+
+[1] J. Alex, L. Benedetti, J. Copp, K.V. Gernaey, U. Jeppsson, I. Nopens, M.N. Pons,
+J.P. Steyer and P. Vanrolleghem, "Benchmark Simulation Model no. 1 (BSM1)", 2018
+"""
+
+# Some more inforation about this module
+__author__ = "Andrew Lee"
+
+import pytest
+
+import pyomo.environ as pyo
+from pyomo.network import Arc
+from pyomo.util.check_units import assert_units_consistent
+
+from idaes.core import FlowsheetBlock
+from idaes.generic_models.unit_models import CSTR
+from idaes.core.util import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.core.util.initialization import propagate_state
+
+from watertap.property_models.activated_sludge.asm1_properties import ASM1ParameterBlock
+from watertap.property_models.activated_sludge.asm1_reactions import (
+    ASM1ReactionParameterBlock,
+)
+
+
+@pytest.mark.integration
+def test_ASM1_reactor():
+    m = pyo.ConcreteModel()
+
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+
+    m.fs.props = ASM1ParameterBlock()
+    m.fs.rxn_props = ASM1ReactionParameterBlock(
+        default={"property_package": m.fs.props}
+    )
+
+    m.fs.R1 = CSTR(
+        default={
+            "property_package": m.fs.props,
+            "reaction_package": m.fs.rxn_props,
+        }
+    )
+    m.fs.R2 = CSTR(
+        default={
+            "property_package": m.fs.props,
+            "reaction_package": m.fs.rxn_props,
+        }
+    )
+    m.fs.R3 = CSTR(
+        default={
+            "property_package": m.fs.props,
+            "reaction_package": m.fs.rxn_props,
+        }
+    )
+    m.fs.R4 = CSTR(
+        default={
+            "property_package": m.fs.props,
+            "reaction_package": m.fs.rxn_props,
+        }
+    )
+    m.fs.R5 = CSTR(
+        default={
+            "property_package": m.fs.props,
+            "reaction_package": m.fs.rxn_props,
+        }
+    )
+
+    # Link units
+    m.fs.stream1 = Arc(source=m.fs.R1.outlet, destination=m.fs.R2.inlet)
+    m.fs.stream2 = Arc(source=m.fs.R2.outlet, destination=m.fs.R3.inlet)
+    m.fs.stream3 = Arc(source=m.fs.R3.outlet, destination=m.fs.R4.inlet)
+    m.fs.stream4 = Arc(source=m.fs.R4.outlet, destination=m.fs.R5.inlet)
+    pyo.TransformationFactory("network.expand_arcs").apply_to(m)
+
+    assert_units_consistent(m)
+
+    # Feed conditions based on manual mass balance of inlet and recycle streams
```

Comment:
What does "manual" mass balance mean? Do we need to calculate the flowrates of the recycle streams separately?